Marlin/src/HAL/HAL_STM32F1/persistent_store_sdcard.cpp

```diff
@@ -66,7 +67,8 @@ static char HAL_STM32F1_eeprom_content[HAL_STM32F1_EEPROM_SIZE];
     SdFile file, root = card.getroot();
     int16_t bytes_written = 0;
     if (file.open(&root, EEPROM_FILENAME, O_CREAT | O_WRITE | O_TRUNC)) {
-      bytes_written = file.write(HAL_STM32F1_eeprom_content, HAL_STM32F1_EEPROM_SIZE);
+      for (int16_t i = 0; i < HAL_STM32F1_EEPROM_SIZE; i++)
+        bytes_written += file.write(HAL_STM32F1_eeprom_content[i]);
```

Comment:
Why do you suppose this isn't working? I see that still isn't checking for an error. The `file.write` function will returns `-1` in the case of an error. Otherwise it returns the value sent in the second argument (1 is the default). Is there something wrong with the `file.write` that is preventing it from writing the bytes, or causing it to return a -1 error value?

Reply:
Well, I don't know enough about the details to provide much advice. I was getting the -1 error, and tracking this it came from a "data misaligned" section in the SDIO driver.

Reply:
it was failing here:   

Marlin\src\HAL\HAL_STM32F1\HAL_sdio_STM32F1.cpp
`if ((0x03 & (uint32_t)data)) return false; // misaligned data`

That seems to be looking to see if things are in 4 bit divisible blocks, but I couldn't figure out why this was not the case when the data size is 4096??

Reply:
It's interesting that the alignment matters in a file write. The align fix PR is about to get merged, so maybe that will be all that's needed…?

Reply:
Ill test it once I get home.